packages/runed/src/lib/utilities/watch/watch.svelte.ts

```diff
@@ -82,23 +83,23 @@ function runWatcher<T>(
 		});
 	});
 
-	$effect(() => {
-		return cleanupRoot;
-	});
+	safelyCleanup(cleanupRoot);
```

Comment:
I really disagree with making `watch` not just use an `$effect`. This would cause a memory leak if you use it outside of component init rather than error.

By explicitly forcing you to use $effect.root, you're reminded to manually cleanup instead of silently leaking.
```ts
const cleanup = $effect.root(() => {
  watch(...)
});
```